firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/Personalization.java

```diff
@@ -17,21 +17,37 @@
 import android.os.Bundle;
 import androidx.annotation.NonNull;
 import com.google.firebase.analytics.connector.AnalyticsConnector;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 import org.json.JSONObject;
 
 public class Personalization {
   public static final String ANALYTICS_ORIGIN_PERSONALIZATION = "fp";
-  public static final String ANALYTICS_PULL_EVENT = "_fpc";
-  public static final String ARM_KEY = "_fpid";
-  public static final String ARM_VALUE = "_fpct";
-  static final String PERSONALIZATION_ID = "personalizationId";
+
+  public static final String ANALYTICS_PULL_EVENT = "personalization_assignment";
+  public static final String ARM_KEY = "arm_key";
+  public static final String ARM_VALUE = "arm_value";
+  public static final String PERSONALIZATION_ID = "personalizationId";
+  public static final String PERSONALIZATION_ID_KEY = "personalization_id";
+  public static final String ARM_INDEX = "armIndex";
+  public static final String ARM_INDEX_KEY = "arm_index";
+  public static final String GROUP = "group";
+
+  public static final String ANALYTICS_PULL_EVENT_INTERNAL = "_fpc";
+  public static final String CHOICE_ID = "choiceId";
+  public static final String CHOICE_ID_KEY = "_fpid";
 
   /** The app's Firebase Analytics client. */
   private final AnalyticsConnector analyticsConnector;
 
+  /** A map of Remote Config parameter key to Personalization ID. */
+  private final Map<String, String> armsCache;
```

Comment:
Looks like this is used to prevent logging the same key and personalization id combination twice? Do we expect the personalization id to change for a particular key? Maybe we can rename the variable to reflect the usage.

Reply:
Oh, that's actually supposed to be choice ID. I think I forgot to change it when we were discussing what fields should be called. We do expect choice IDs to change, and we need to log each at least once. There will probably some de-duping on the back-end too, but this is just a quick way to reduce some of that work.